cpp/src_prims/sparse/distance/l2_distance.cuh

```diff
@@ -176,7 +176,7 @@ class l2_sqrt_expanded_distances_t
       out_dists, out_dists, this->config_->a_nrows * this->config_->b_nrows,
       [p] __device__(value_t input) {
         int neg = input < 0 ? -1 : 1;
-        return powf(fabs(input), p) * neg;
+        return pow(std::abs(input), p) * neg;
```

Comment:
The cuda math library has optimized functions for `abs`. If you call the host `std::abs` function (which is allowed because we set a compiler flag to enable it) then I don't believe the cuda library is smart enough to change that out for the cuda optimized version. Same with the one below.